linetools/isgm/abssurvey.py

```diff
@@ -0,0 +1,325 @@
+        tree : str, optional
+          Path to .dat files
+        kwargs :
+          Passed to __init__
+        """
+        if tree is None:
+            tree = ''
+        # Load up (if possible)
+        data = ascii.read(tree+flist, data_start=0,
+                          guess=False,format='no_header')
+        slf = cls(nsys=len(data),**kwargs)
+        slf.tree = tree
+        slf.flist = flist
+
+        # Load up
+        slf.dat_files = list(data['col1'])
+        print('Read {:d} files from {:s} in the tree {:s}'.format(
+            slf.nsys, slf.flist, slf.tree))
+        # Generate AbsSys list
+        for dat_file in slf.dat_files:
+            slf._abs_sys.append(set_absclass(slf.abs_type).from_datfile(dat_file,tree=slf.tree))
+
+        return slf
+
+    @classmethod
+    def from_sfits(cls, summ_fits, **kwargs):
+        """Generate the Survey from a summary FITS file
+
+        Handles SPEC_FILES too.
+        """
+        # Read
+        systems = QTable.read(summ_fits)
+        slf = cls(nsys=len(systems),**kwargs)
+        # Dict
+        kdict = dict(NHI=['NHI','logNHI'],
+            sig_NHI=['sig(logNHI)'],
+            name=['Name'],
+            vlim=['vlim'],
+            zabs=['Z_LLS'],
+            zem=['Z_QSO'],
+            RA=['RA'],
+            Dec=['DEC','Dec'])
+        # Parse the Table
+        inputs = {}
+        for key in kdict.keys():
+            vals, tag = lsio.get_table_column(kdict[key],[systems],idx=0)
+            if vals is not None:
+                inputs[key] = vals
+        # vlim
+        if not 'vlim' in inputs.keys():
+            default_vlim = [-500,500.]*u.km/u.s
+            inputs['vlim'] = [default_vlim]*slf.nsys
+        # Generate
+        for kk in range(slf.nsys):
+            # Generate keywords
+            kwargs = {}
+            args = {}
+            for key in inputs.keys():
+                if key in ['vlim','zabs','RA','Dec']:
+                    args[key] = inputs[key][kk]
+                else:
+                    kwargs[key] = inputs[key][kk]
+            # Instantiate
+            abssys = set_absclass(slf.abs_type)((args['RA'],args['Dec']), args['zabs'], args['vlim'], **kwargs)
+            # spec_files
+            try:
+                abssys.spec_files += systems[kk]['SPEC_FILES'].tolist()
+            except (KeyError,AttributeError):
+                pass
+            slf._abs_sys.append(abssys)
+        # Return
+        return slf
+
+    def __init__(self, abs_type, nsys=0, ref=''):
+        # Expecting a list of files describing the absorption systems
+        """  Initiator
+
+        Parameters
+        ----------
+        abs_type : str
+          Type of AbsSystem in the Survey, e.g.  MgII, DLA, LLS
+        nsys : int, optional
+          Number of AbsSystem in the Survey
+        ref : string, optional
+          Reference(s) for the survey
+        """
+        self.abs_type = abs_type
+        self.ref = ref
+        self._abs_sys = []
```

Comment:
But in the example notebook you access it directly to add new systems. If that's not the intended usage, the example should be changed.  (Also `_abs_sys`  is way better than `_abssys`)